jme3-core/src/main/resources/Common/ShaderLib/PBRLighting.glsllib

```diff
@@ -0,0 +1,184 @@
+#import "Common/ShaderLib/Lighting.glsllib"
+#import "Common/ShaderLib/PBR.glsllib"
+
+//declare PBR Lighting vars
+uniform vec4 g_LightData[NB_LIGHTS];
+uniform vec3 g_CameraPosition;
+uniform vec4 g_AmbientLightColor;
+
+#if NB_PROBES >= 1
+    uniform samplerCube g_PrefEnvMap;
+    uniform vec3 g_ShCoeffs[9];
+    uniform mat4 g_LightProbeData;
+#endif
+#if NB_PROBES >= 2
+    uniform samplerCube g_PrefEnvMap2;
+    uniform vec3 g_ShCoeffs2[9];
+    uniform mat4 g_LightProbeData2;
+#endif
+#if NB_PROBES == 3
+    uniform samplerCube g_PrefEnvMap3;
+    uniform vec3 g_ShCoeffs3[9];
+    uniform mat4 g_LightProbeData3;
+#endif
+
+// Specular-AA
+#ifdef SPECULAR_AA_SCREEN_SPACE_VARIANCE
+    uniform float m_SpecularAASigma;
+#endif
+#ifdef SPECULAR_AA_THRESHOLD
+    uniform float m_SpecularAAKappa;
+#endif
+

```

Comment:
These uniforms should be relegated to PBRLightingParamsReader since they are specific to PBRLighting material params

Reply:
Yes I must've made a mistake leaving the SpecularAA params there.

As for the other lighting variables:
`uniform vec4 g_LightData[NB_LIGHTS];
uniform vec3 g_CameraPosition;
uniform vec4 g_AmbientLightColor;

#if NB_PROBES >= 1
    uniform samplerCube g_PrefEnvMap;
    uniform vec3 g_ShCoeffs[9];
    uniform mat4 g_LightProbeData;
#endif
#if NB_PROBES >= 2
    uniform samplerCube g_PrefEnvMap2;
    uniform vec3 g_ShCoeffs2[9];
    uniform mat4 g_LightProbeData2;
#endif
#if NB_PROBES == 3
    uniform samplerCube g_PrefEnvMap3;
    uniform vec3 g_ShCoeffs3[9];
    uniform mat4 g_LightProbeData3;
#endif`

Would these just be better off placed in the .frag file then (rather than either .glsllib), since they can be used by both a single pass and multi-pass implementation?

Reply:
This goes together with the other comment, unless the logic is abstracted,  there isn't a good place to declare uniforms.

When i said spaghetti code i was referring to this: let's say i want to fork PBRLighting.frag, i need to know that this glsllib declares these uniforms, but also the other one declares some others, so if my fork needs the g_CameraPosition, and i have this glsllib included, i need to know that i can't declare the g_CameraPosition uniform again, because it is already in the glsllib. 

Imagine this applied to all the shader code, now you have libraries that they look as if they provided useful reusable code, but they actually can conflict unless used into a specific way. 

This can be at best a temporary fix for simplify PBRLighting forks, as you said, but it is not the right way to make it modular.

Reply:
I had planned to declare all potentially reusable uniforms in the .frag shader to avoid any potential conflicting uniforms in a glsllib that needs that variable.

There may be some that I missed, but I did,(for example) keep varyings and uniforms like wPosition and tbnMat declared in the .frag shader specifically because I thought those are likelky to be re-used in other .glsllibs. 

However I just wasn't certain if these lighting params were the type that would ever need declared again, since I don't have a ton of experience with other lighting code aside from whats in PBRLighting.frag curently.

But I can move all of those uniforms you mentioned back to the .frag shader.

And yes this PR is probably more of a temporary fix, or morseo only a partial modularization, mostly aimed at modularizing the param & texture reads more than the lighting, since I don't have as much experience with the lighting side. And as I mentioned in the other comment, I don't see as big of a need to modularize the lighting on core when we only have one current lighting pipeline on core at the moment anyways. But that certainly is the correct solution at some point in the future.